app/components/slack-context-menu.js

```diff
@@ -0,0 +1,165 @@
+(function () {
+	'use strict';
+	// Load in our dependencies
+	var gui = window.require('nw.gui');
+	var clipboard = gui.Clipboard.get();
+
+	var GOOGLE_SEARCH_URL = "https://www.google.com/#q=";
+
+	function isSelected(target) {
+		return target.selectionStart != target.selectionEnd;
+	}
+
+	function getSelectedValue(target) {
+		return target.value.substring(target.selectionStart, target.selectionEnd);
+	}
+
+	function removeSelectedValue(target) {
+		var selectionStart = target.selectionStart;
+		var selectedValue = target.value.substring(target.selectionStart, target.selectionEnd);
+		target.value = target.value.substring(0, target.selectionStart) + target.value.substring(target.selectionEnd);
```

Comment:
Most people usually get confused between `substr` and `substring`. Can we move to `slice` to make it more array-like?

http://www.danvk.org/2014/11/17/javascript-string-splice-substr-substring-which-to-use.html